corfu-prescient.el

```diff
@@ -0,0 +1,196 @@
+  "Enable or disable `prescient-completion-local-mode' in `corfu-mode-hook'."
+  (if corfu-mode
+      (progn
+        (unless prescient-completion-local-mode
+          (prescient-completion-local-mode 1)
+          (setq corfu-prescient--local-settings t)))
+    (when corfu-prescient--local-settings
+      (prescient-completion-local-mode -1)
+      (setq corfu-prescient--local-settings nil))))
+
+;;;###autoload
+(define-minor-mode corfu-prescient-mode
+  "Minor mode to use prescient.el in Corfu menus.
+
+This mode will:
+- if `corfu-prescient-override-sorting' is non-nil,
+  configure `corfu-sort-override-function' and set
+ `corfu-prescient-enable-filtering' to t
+
+- if `corfu-prescient-enable-filtering' is non-nil,
+  configure `corfu-sort-function'
+
+- if `corfu-prescient-enable-filtering' is non-nil:
+  - enable `prescient-completion-local-mode' in the Corfu buffer
+  - bind `prescient-toggle-map' to `M-s' in `corfu-map'
+
+- advise `corfu-insert' to remember candidates"
+  :global t
+  :group 'prescient
+  (if corfu-prescient-mode
+      ;; Turn on the mode.
+      (progn
+        ;; Prevent messing up variables if we explicitly enable the
+        ;; mode when it's already on.
+        (corfu-prescient-mode -1)
+        (setq corfu-prescient-mode t)
+
+        (when corfu-prescient-override-sorting
+          (setq corfu-prescient-enable-sorting t)
+          (cl-shiftf corfu-prescient--old-sort-override-function
+                     corfu-sort-override-function
+                     #'prescient-completion-sort))
+
+        (when corfu-prescient-enable-sorting
+          (cl-shiftf corfu-prescient--old-sort-function
+                     corfu-sort-function
+                     #'prescient-completion-sort))
+
+        (when corfu-prescient-enable-filtering
+          ;; Configure completion settings.
+          (add-hook 'corfu-mode-hook
+                    #'corfu-prescient--change-completion-settings)
+
+          ;; Bind toggling commands.
+          (setq corfu-prescient--old-toggle-binding
+                (lookup-key corfu-map (kbd "M-s")))
+          (define-key corfu-map (kbd "M-s") prescient-toggle-map)
+
+          ;; Make sure Corfu refreshes immediately.
+          (add-hook 'prescient--toggle-refresh-functions
+                    #'corfu-prescient--toggle-refresh)
+
+          ;; Clean up the local versions of the toggling variables
+          ;; after the Corfu pop-up closes. For the toggling vars, it
+          ;; is the commands themselves that make the variables buffer
+          ;; local.
+          (cl-callf cl-union corfu--state-vars prescient--toggle-vars
+                    :test #'eq))
+
+        ;; While sorting might not be enabled in Corfu, it might
+        ;; still be enabled in another UI, such as Selectrum or Vertico.
+        ;; Therefore, we still want to remember candidates.
+        (advice-add 'corfu-insert :after #'corfu-prescient--remember))
```

Comment:
I think it would feel more consistent to only install the advice if sorting is enabled.

Reply:
Maybe, but because the candidate selection should affect sorting in other UIs using `prescient.el`, I think that the advice should always be added.